src/sqlancer/postgres/PostgresProvider.java

```diff
@@ -218,24 +220,51 @@ protected TestOracle getTestOracle(PostgresGlobalState globalState) throws SQLEx
 
     @Override
     public Connection createDatabase(PostgresGlobalState globalState) throws SQLException {
-        String url = "jdbc:postgresql://localhost:5432/test";
+        String username = globalState.getOptions().getUserName();
+        String password = globalState.getOptions().getPassword();
+        String entryPath = "test";
+        String entryURL = globalState.getDmbsSpecificOptions().connectionURL;
+        // trim URL to exclude "jdbc:"
+        if (entryURL.startsWith("jdbc:")) {
+            entryURL = entryURL.substring(5);
+        }
+        try {
+            URI uri = new URI(entryURL);
+            String userInfoURI = uri.getUserInfo();
+            String pathURI = uri.getPath();
+            if (userInfoURI != null) {
+                // username and password specified in URL take precedence
+                String[] userInfo = userInfoURI.split(":", 2);
+                username = userInfo[0];
+                password = userInfo[1];
+                int userInfoIndex = entryURL.indexOf(userInfoURI);
+                entryURL = entryURL.substring(0, userInfoIndex)
+                        + entryURL.substring(userInfoIndex + userInfoURI.length() + 1);
+            }
+            if (pathURI != null) {
+                entryPath = pathURI;
+            }
+        } catch (URISyntaxException e) {
+            throw new AssertionError(e);
+        }
         String databaseName = globalState.getDatabaseName();
-        Connection con = DriverManager.getConnection(url, globalState.getOptions().getUserName(),
-                globalState.getOptions().getPassword());
-        globalState.getState().logStatement("\\c test;");
```

Comment:
Do we need to remove this? Logging a `\c test` is useful when reducing a test case manually, since changing to the default database `test` makes it possible to drop the current database of the test case.